src/annotator/config/settings.ts

```diff
@@ -99,16 +103,29 @@ export function settingsFrom(window_: Window): SettingsGetters {
     }
   }
 
+  function sideBySide(): SideBySideOptions {
+    const value = hostPageSetting('sideBySide');
+
+    return {
+      mode:
+        // The sideBySide prop has to be an object.
+        // The mode prop has to be 'auto' or 'manual', if defined.
+        !isObject(value) || !('mode' in value) || !isSideBySideMode(value.mode)
+          ? 'auto'
+          : value.mode,
```

Comment:
We could do with some more declarative way of expressing schemas and validating the config in future.

Reply:
Yep. I think some validation library could be useful here.